FordLabs/retroquest
# Make WebSocket Unidirectional

## Overview
Removes the websocket update handlers and replaces them with the appropriate REST calls instead. We've noticed that people sometimes have their thoughts and action items disappear when entered. Some of that can be attributed to when a tab is left open too long a token will fail the auth check causing the message to be swallowed. Given the security overhaul of the backend in order to make the websocket publish work correctly with auth and with there being some unresolved issues with the websocket, we decided to replace the publish events over the websocket with REST calls.

This pull request makes existing REST alternatives publish events to the appropriate topics on the websocket when called and removes the MessageMappings from the relevant controllers.

Connects #351 

### Notes
In order to handle auth on the websocket we would've needed to have an additional security configuration to handle both the authentication and authorization of messages over the websocket. We would also have had to add a user specific topic that is listened to on the frontend so it can receive auth errors and react accordingly. 

We still need to implement a guarded httpClient that redirects to the login page on an auth error, but this pull request makes that much easier to implement.

## Testing Instructions
This change shouldn't be noticable from a user perspective, except delivery should be more consistent. Will still not work with an expired token.
* Build the prod deployment of the application.
* Start a new retro and join from two different browsers (ideally two different computers)
* Conduct retro like normal, making sure to use all functions (edit, like thought, change column title, edit action items, delete stuff, etc.)
* End retro and check archived items
* Everything should work as normal
